Sources/SolToBoogie/GhostVarAndAxiomGenerator.cs

```diff
@@ -364,9 +397,52 @@ private BoogieAxiom GenerateAbiEncodePackedAxiomTwoArgs()
             var qExpr = new BoogieQuantifiedExpr(true, new List<BoogieIdentifierExpr>() { qVar11, qVar12, qVar21, qVar22 }, 
                 new List<BoogieType>() { BoogieType.Int, BoogieType.Int, BoogieType.Int, BoogieType.Int }, bodyExpr, triggers);
 
+            return new BoogieAxiom(qExpr);
+        }
+
+
+        private BoogieAxiom GenerateAbiEncodePackedAxiomOneArgRef()
+        {
+
+            var qVar1 = QVarGenerator.NewQVar(0, 0);
+            var qVar2 = QVarGenerator.NewQVar(0, 1);
+            var eqVar12 = new BoogieBinaryOperation(BoogieBinaryOperation.Opcode.EQ, qVar1, qVar2);
+            var qVar1Func = new BoogieFuncCallExpr("abiEncodePacked1R", new List<BoogieExpr>() { qVar1 });
+            var qVar2Func = new BoogieFuncCallExpr("abiEncodePacked1R", new List<BoogieExpr>() { qVar2 });
+            var eqFunc12 = new BoogieBinaryOperation(BoogieBinaryOperation.Opcode.NEQ, qVar1Func, qVar2Func);
+            var bodyExpr = new BoogieBinaryOperation(BoogieBinaryOperation.Opcode.OR, eqVar12, eqFunc12);
+            var triggers = new List<BoogieExpr>() { qVar1Func, qVar2Func };
+
+            // forall q1:int, q2:int :: q1 == q2 || abiEncodePacked(q1) != abiEncodePacked(q2) 
+            var qExpr = new BoogieQuantifiedExpr(true, new List<BoogieIdentifierExpr>() { qVar1, qVar2 }, new List<BoogieType>() { BoogieType.Ref, BoogieType.Ref}, bodyExpr, triggers);
+
             return new BoogieAxiom(qExpr);
         }
 
+        private BoogieAxiom GenerateAbiEncodePackedAxiomTwoArgsOneRef()
+        {
+            var qVar11 = QVarGenerator.NewQVar(0, 0);
+            var qVar12 = QVarGenerator.NewQVar(0, 1);
+            var qVar21 = QVarGenerator.NewQVar(1, 0);

```

Comment:
did you already branch out of your other PR?

Reply:
No, the problem was that the ABIEncode additions were in my main branch, so they appeared in all the branches I created afterwards 🤦‍♂